javascript/ql/src/experimental/Security/CWE-090/LdadInjection.ql

```diff
@@ -0,0 +1,20 @@
+/**
+ * @name LDAP query built from user-controlled sources
+ * @description Building an LDAP query from user-controlled sources is vulnerable to insertion of
+ *              malicious LDAP code by the user.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id javascript/ldap-injection
+ * @tags security
+ *       external/cwe/cwe-090
+ */
+
+import javascript
+import DataFlow::PathGraph
+import LdapInjection::LdapInjection
+
+from LdapInjectionConfiguration config, DataFlow::PathNode source, DataFlow::PathNode sink
+where config.hasFlowPath(source, sink)
+select sink.getNode(), source, sink, "LDAP query might include code from $@.", source.getNode(),
+  "user-provided value"
```

Comment:
```suggestion
select sink.getNode(), source, sink, "$@ might include code from $@.", sink.getNode().(Sink).getQueryCall(), "LDAP query call", source.getNode(),
  "user-provided value"
```

Currently your query highlights the write to `filter` when running the query. Maybe you could add a `getHighlight()`/`getQueryCall()` to the sink, and use that to highlight the call to `search()`.  
The suggestion is how the `select` would end up looking.

Reply:
Yes, it makes sense and also it is a good opportunity for me to learn something new. I tried to highlight the call to `search()` (or any other client API call), but probably there are better way to accomplish it. Any feedback is welcome